lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1660,6 +1660,78 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// This pass iteratively checks the use count of each value, and reformats the
+// MLIR such that every value has exact one use. If a value has multiple uses, a
+// fork is inserted. If a value has no use, a sink is inserted.
+void insertOpForExactOneUse(Value value, handshake::FuncOp funcOp,
+                            OpBuilder &rewriter) {
```

Comment:
If you're gonna be setting the insertion point, I wouldn't pass in a shared `OpBuilder`. They're pretty trivial to build -- AFIACT, it's just a stack allocation and setting the insertion point, which you're already doing.